HPXMLOutputReport/measure.rb

```diff
@@ -0,0 +1,418 @@
+      end
+    elsif cost_mult_type == 'Duct Unconditioned Surface Area'
+      hpxml.hvac_distributions.each do |hvac_distribution|
+        hvac_distribution.ducts.each do |duct|
+          next if [HPXML::LocationLivingSpace, HPXML::LocationBasementConditioned].include?(duct.duct_location)
+
+          cost_mult += duct.duct_surface_area
+        end
+      end
+    elsif cost_mult_type.include? 'Size Heating System'
+      hpxml.heating_systems.each do |heating_system|
+        next if heating_system.id != cost_mult_type.split(': ')[1]
+
+        cost_mult += UnitConversions.convert(heating_system.heating_capacity, 'btu/hr', 'kbtu/hr')
+      end
+
+      hpxml.heat_pumps.each do |heat_pump|
+        next if heat_pump.id != cost_mult_type.split(': ')[1]
+
+        cost_mult += UnitConversions.convert(heat_pump.heating_capacity, 'btu/hr', 'kbtu/hr')
+      end
+    elsif cost_mult_type.include? 'Size Cooling System'
+      hpxml.cooling_systems.each do |cooling_system|
+        next if cooling_system.id != cost_mult_type.split(': ')[1]
+
+        cost_mult += UnitConversions.convert(cooling_system.cooling_capacity, 'btu/hr', 'kbtu/hr')
+      end
+
+      hpxml.heat_pumps.each do |heat_pump|
+        next if heat_pump.id != cost_mult_type.split(': ')[1]
+
+        cost_mult += UnitConversions.convert(heat_pump.cooling_capacity, 'btu/hr', 'kbtu/hr')
+      end
+    elsif cost_mult_type.include? 'Size Heat Pump Backup'
+      hpxml.heat_pumps.each do |heat_pump|
+        next if heat_pump.id != cost_mult_type.split(': ')[1]
+
+        cost_mult += UnitConversions.convert(heat_pump.backup_heating_capacity, 'btu/hr', 'kbtu/hr')
+      end
+    elsif cost_mult_type == 'Size Water Heater'
+      hpxml.water_heating_systems.each do |water_heating_system|
+        next if water_heating_system.tank_volume.nil?
+
+        cost_mult += water_heating_system.tank_volume
+      end
+    elsif cost_mult_type == 'Flow Rate Mechanical Ventilation'
+      hpxml.ventilation_fans.each do |ventilation_fan|
+        next unless ventilation_fan.used_for_whole_building_ventilation
+
+        if ventilation_fan.rated_flow_rate
+          cost_mult += ventilation_fan.rated_flow_rate
+        end
+      end
+    elsif cost_mult_type == 'Slab Perimeter Exposed Conditioned'
+      hpxml.slabs.each do |slab|
+        next unless slab.is_exterior_thermal_boundary
+
+        cost_mult += slab.exposed_perimeter
+      end
+    elsif cost_mult_type == 'Rim Joist Area Above-Grade Exterior'
+      hpxml.rim_joists.each do |rim_joist|
+        cost_mult += rim_joist.area
+      end
+    end
+    return cost_mult
+  end
+
+  def assign_primary_and_secondary(hpxml, cost_multipliers)
+    has_primary_heating_system = false
+    hpxml.heating_systems.each do |heating_system|
+      next if !heating_system.primary_system
+
+      cost_mult = BaseOutput.new
+      cost_mult.output = cost_multipliers["#{BS::SizeHeatingSystem}: #{heating_system.id}"].output
```

Comment:
I've realized we are being pretty inconsistent between HVAC systems (reported by system) vs water heaters or mechanical ventilation (summed across systems).

I know I had suggested otherwise, but for this first cut, let's just stick with simple results until it's requested that we provide more detail. So let's just report A) the sum of HVAC systems for heating/cooling sizes (i.e., not outputs by system), and B) the primary/secondary sizes when applicable.